parser_test.go

```diff
@@ -292,24 +112,32 @@ func TestParserChechChunks(t *testing.T) {
 		"/{id}/{name}",
 		"/{id}/{name}/",
 		"/{id}-{name}/",
+		"/{id:[0-9]+}/",
+		"/path1/{id:/ab+c/}",
+		"/path1/{id:/d(b+)d/g}/path2",
+		"/{id:[0-9]+}-{name:/ab+c/}/",
 	}
 
 	expectedChunks := []string{
-		"[{0 /}]",
-		"[{0 /path1}]",
-		"[{0 /path1/path2}]",
-		"[{0 /path1/path2/}]",
-		"[{0 /path1/} {1 id}]",
-		"[{0 /path1/} {1 id} {0 /}]",
-		"[{0 /path1/} {1 id} {0 /path2}]",
-		"[{0 /} {1 id}]",
-		"[{0 /} {1 id} {0 /}]",
-		"[{0 /} {1 id} {0 /path1}]",
-		"[{0 /asdf-} {1 id}]",
-		"[{0 /} {1 id} {0 -adfasf}]",
-		"[{0 /} {1 id} {0 /} {1 name}]",
-		"[{0 /} {1 id} {0 /} {1 name} {0 /}]",
-		"[{0 /} {1 id} {0 -} {1 name} {0 /}]",
+		"[{0 / }]",
+		"[{0 /path1 }]",
+		"[{0 /path1/path2 }]",
+		"[{0 /path1/path2/ }]",
+		"[{0 /path1/ } {1 id }]",
+		"[{0 /path1/ } {1 id } {0 / }]",
+		"[{0 /path1/ } {1 id } {0 /path2 }]",
+		"[{0 / } {1 id }]",
+		"[{0 / } {1 id } {0 / }]",
+		"[{0 / } {1 id } {0 /path1 }]",
+		"[{0 /asdf- } {1 id }]",
+		"[{0 / } {1 id } {0 -adfasf }]",
+		"[{0 / } {1 id } {0 / } {1 name }]",
+		"[{0 / } {1 id } {0 / } {1 name } {0 / }]",
+		"[{0 / } {1 id } {0 - } {1 name } {0 / }]",
+		"[{0 / } {1 id ^[0-9]+$} {0 / }]",
+		"[{0 /path1/ } {1 id ^/ab+c/$}]",
+		"[{0 /path1/ } {1 id ^/d(b+)d/g$} {0 /path2 }]",
```

Comment:
Are you sure this is what we expect regarding the start/ending of the regular expresion? I mean, the "/g" is a modifier isn't it? So I think the correct regex should be /^d(b+)d$/g. Have you checked this kind of regular expresion against some string? Let's talk about this